### PR TITLE
Added Repo for Providers. And the int tests needed

### PR DESCRIPTION
**Pull request by Pauljgraham**

### CODE REVIEW

File: src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/DbSetupScripts/6_PeriodEnd.Deds.PaymentsDue.Add_APIIndexes.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
+IF NOT EXISTS(SELECT NULL FROM sys.indexes WHERE name='IX_RequiredPayments_AccountId')
```
Comment (carlbm) [inline src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/DbSetupScripts/6_PeriodEnd.Deds.PaymentsDue.Add_APIIndexes.sql:1]: Should this file be part of what we're doing?

Comment (carlbm) [reply src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/DbSetupScripts/6_PeriodEnd.Deds.PaymentsDue.Add_APIIndexes.sql:1]: Actually... did you add this file instead of adding a link?

Comment (carlbm) [reply src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/DbSetupScripts/6_PeriodEnd.Deds.PaymentsDue.Add_APIIndexes.sql:1]: Same comment for other sql files. I can't comment on them directly

File: src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/DbSetupScripts/Summarisation.Deds.DML.sql [hunk 2]
```diff
@@ -0,0 +1,17 @@
+TRUNCATE TABLE [dbo].[Collection_Period_Mapping]
+GO
+
```
Comment (carlbm) [inline src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/DbSetupScripts/Summarisation.Deds.DML.sql:3]: Could be worth using 1819 instead of 1617?

**Review by carlbm — CHANGES_REQUESTED** [review]

File: src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/DbSetupScripts/6_PeriodEnd.Deds.PaymentsDue.Add_APIIndexes.sql (hunk 1 above)
Comment (Pauljgraham) [reply src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/DbSetupScripts/6_PeriodEnd.Deds.PaymentsDue.Add_APIIndexes.sql:1]: This is a direct copy from existing paymentdue integration test. Whilst I totally agree there is great value in re-working the integration test infrastructure and coming up with a good initialisation pattern, I think this should be done for all the int tests.

File: src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/DbSetupScripts/Summarisation.Deds.DML.sql (hunk 2 above)
Comment (Pauljgraham) [reply src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/DbSetupScripts/Summarisation.Deds.DML.sql:3]: Probably not going to use this file at all. But am very nervous about changing sql scripts or removing scripts as they are all so inter-related

File: src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/Tests/Infrastructure/GivenAProvidersRepository.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            result.Count().Should().Be(2);
+        }
+
+        [Test]
+        public void ThenTheUkprnsMatch()
+        {
+            var result = _sut.GetAllProviders().OrderBy(x=>x.Ukprn).ToList();
+            result.First().Ukprn.Should().Be(1);
+            result.Last().Ukprn.Should().Be(2);
+        }
```
Comment (scottywakefield) [inline src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/Tests/Infrastructure/GivenAProvidersRepository.cs]: You are missing some tests.. 
should return an empty list if no records found.
should return correct IlrSubmissionDateTime 

ps: use the layout of the integration tests in payments due as a template.

**Review by scottywakefield — CHANGES_REQUESTED** [review]

File: src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/Tests/Infrastructure/GivenAProvidersRepository.cs (hunk 3 above)
Comment (Pauljgraham) [reply src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/Tests/Infrastructure/GivenAProvidersRepository.cs]: In this case I don't think we need to test for Empty Records as there is no where clause and we don't need to test the Dapper layer. 

As for IlrSubmissionDateTime, we are not using this field, but I'm happy to test a value is mapped

Comment (scottywakefield) [reply src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/Tests/Infrastructure/GivenAProvidersRepository.cs]: I mention the empty list because it has a big impact if you return null instead of an empty list.

File: src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/DbSetupScripts/6_PeriodEnd.Deds.PaymentsDue.Add_APIIndexes.sql (hunk 1 above)
Comment (carlbm) [reply src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/DbSetupScripts/6_PeriodEnd.Deds.PaymentsDue.Add_APIIndexes.sql:1]: It's still wrong though. It should be linked so that changes to the source are reflected by changes to the integration tests. Otherwise we're just asking for a very subtle bug down the line when we change something and the tests continue to work when they shouldn't

**Review by carlbm — CHANGES_REQUESTED** [review]

File: src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/DbSetupScripts/6_PeriodEnd.Deds.PaymentsDue.Add_APIIndexes.sql (hunk 1 above)
Comment (Pauljgraham) [reply src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/DbSetupScripts/6_PeriodEnd.Deds.PaymentsDue.Add_APIIndexes.sql:1]: Carl this problem exists in PaymentsDueIntegration Tests and the other integration tests. The whole integration testframework needs to be reviewed.

**Review by scottywakefield — APPROVED** [review]
